doc/design/listener.md

```diff
@@ -108,20 +108,34 @@ which event is attached, like in the _WaitSet_.
 | |   void executeCallback()   |                  |      | 1
 | |   bool reset()             |                  |      |
 | |   bool init(...)           |                  |      | n
-| |                            |                  | +----------------------------------------------------+
-| |   - m_origin               |                  | | Triggerable (e.g. Subscriber)                      |
-| |   - m_callback             |                  | |                                                    |
-| |   - m_invalidationCallback |                  | |   void invalidateTrigger(const uint64_t triggerId) |
-| |   - m_eventId              |                  | |   void enableEvent(TriggerHandle&&)                |
-| +----------------------------+                  | |   void disableEvent(const EventEnum event)         |
-+-------------------------------------------------+ |                                                    |
-                                                    |   - m_triggerHandle : TriggerHandle                |
-                                                    +----------------------------------------------------+
+| |                            |                  | +-------------------------------------------------------+
```

Comment:
You can now use also plantuml on the design docs, see https://github.com/eclipse-iceoryx/iceoryx/tree/master/doc/design

Reply:
Thanks for the hint but in this PR I would keep the ASCII diagram